### PR TITLE
Add Kong reverse proxy example

### PR DESCRIPTION
**Pull request by f35f0ef9d0e827dae86552d3899f78fc**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: docs/reverse_proxy.md [hunk 1]
```diff
@@ -66,6 +67,40 @@ server {
...
+
+# Synapse
+services:
+- name: synapse-service
+  url: http://127.0.0.1:8008
+  routes:
+  - name: synapse-client-route
+    paths:
+    - /_matrix
+    - /_synapse/client
```
Comment (anoadragon453) [inline docs/reverse_proxy.md:84]: This also covers `/_matrix/federation/...` routes (server<->server communication), so `synapse-client-route` may be slightly misleading. Perhaps `synapse-public-route` instead?

File: README.rst [hunk 2]
```diff
@@ -182,6 +182,7 @@ Using a reverse proxy with Synapse
 
 It is recommended to put a reverse proxy such as
 `nginx <https://nginx.org/en/docs/http/ngx_http_proxy_module.html>`_,
+`Kong <https://docs.konghq.com/getting-started-guide/2.3.x/overview/>`_,
```
Comment (anoadragon453) [inline README.rst:185]:
```suggestion
`Kong <https://docs.konghq.com/getting-started-guide/latest/overview/>`_,
```

This works as well and wouldn't need to be maintained :slightly_smiling_face:

File: changelog.d/9704.doc [hunk 3]
```diff
@@ -0,0 +1 @@
+Add Kong reverse proxy example.
```
Comment (anoadragon453) [inline changelog.d/9704.doc:1]:
```suggestion
Add Kong reverse proxy config example.
```

File: docs/reverse_proxy.md [hunk 4]
```diff
@@ -66,6 +67,40 @@ server {
...
+    config:
+      allow:
+      - x.x.x.x
+```
+
+**NOTE**: The `ip-restriction` built-in plugin is optional to restrict
+any route (in this case, the admin API) to one or more IP addresses.
+
+**NOTE 2**: The example above is using Kong's
+[DB-less configuration format](https://docs.konghq.com/gateway-oss/2.3.x/db-less-and-declarative-config/).
```
Comment (anoadragon453) [inline docs/reverse_proxy.md:102]:
```suggestion
[DB-less configuration format](https://docs.konghq.com/gateway-oss/latest/db-less-and-declarative-config/).
```

and here.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Thank you for your contribution! This looks good to me on the whole, just a few tweaks as noted below.

**Comment by f35f0ef9d0e827dae86552d3899f78fc** [comment]
> Thank you for your contribution! This looks good to me on the whole, just a few tweaks as noted below.

Thanks for reviewing the PR. I have submitted the requested changes. Makes sense to me!

**Comment by erikjohnston** [comment]
@f35f0ef9d0e827dae86552d3899f78fc Hi, I'm afraid we need you to sign off on the PR with a real name before we can accept the patch.

**Comment by anoadragon453** [comment]
Hey @f35f0ef9d0e827dae86552d3899f78fc, I'm afraid we can't accept this PR without a valid sign-off.

Feel free to re-open with one though!

**Pull request closed without merging**